app/controllers/vm_cloud_controller.rb

```diff
@@ -309,6 +307,90 @@ def cancel_action(message)
     replace_right_cell
   end
 
+  def evacuate
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id]) # Set the VM object
+    if @record.is_available?(:evacuate) && !@record.ext_management_system.nil?
+      drop_breadcrumb(
+        :name => _("Evacuate Instance '%{name}'") % {:name => @record.name},
+        :url  => "/vm_cloud/evacuate"
+      ) unless @explorer
+      @in_a_form = true
+      @refresh_partial = "vm_common/evacuate"
+    else
+      add_flash(_("Unable to evacuate %{instance} \"%{name}\": %{details}") % {
+        :instance => ui_lookup(:table => 'vm_cloud'),
+        :name     => @record.name,
+        :details  => @record.is_available_now_error_message(:evacuate)}, :error)
+    end
+  end
+  alias_method :instance_evacuate, :evacuate
+
+  def evacuate_form_fields
+    assert_privileges("instance_evacuate")
+    @record = find_by_id_filtered(VmOrTemplate, params[:id])
+    hosts = []
+    unless @record.ext_management_system.nil?
+      connection = @record.ext_management_system.connect
+      current_hostname = connection.handled_list(:servers).find do |s|
+        s.name == @record.name
+      end.os_ext_srv_attr_hypervisor_hostname
+      hosts = connection.hosts.select { |h| h.service_name == "compute" && h.host_name != current_hostname }.map do |h|
+        {:name => h.host_name, :id => h.id}
```

Comment:
"id" is not a valid field for a fog compute host. If you replace "h.id" with "h.host_name" (i.e. set name and id both to the same host_name value) it works and I can successfully migrate to a specific destination host.